plugins/filter/to_prettytable.py

```diff
@@ -0,0 +1,380 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2025, Timur Gadiev <tgadiev@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+
+DOCUMENTATION = '''
+name: to_prettytable
+short_description: Format a list of dictionaries as an ASCII table
+version_added: "10.7.0"
+author: Timur Gadiev (@tgadiev)
+description:
+- This filter takes a list of dictionaries and formats it as an ASCII table using the I(prettytable) Python library.
+requirements:
+- prettytable
+options:
+  _input:
+    description: A list of dictionaries to format.
+    type: list
+    elements: dictionary
+    required: true
+  column_order:
+    description: List of column names to specify the order of columns in the table.
+    type: list
+    elements: string
+  header_names:
+    description: List of custom header names to use instead of dictionary keys.
+    type: list
+    elements: string
+  column_alignments:
+    description:
+    - >-
+        Dictionary where keys are column names and values are alignment settings.
+        Valid alignment values are C(left), C(center), C(right), C(l), C(c), or C(r).
+    - >-
+        For example, V({'name': 'left', 'id': 'right'}) will align the C(name) column to the left
+        and the C(id) column to the right.
+    type: dictionary
+'''
+
+EXAMPLES = '''
+---
+- name: Display a list of users as a table
+  vars:
+    users:
+      - name: Alice
+        age: 25
+        role: admin
+      - name: Bob
+        age: 30
+        role: user
+  debug:
+    msg: "{{ users | community.general.to_prettytable }}"
+
+- name: Display a table with custom column ordering
+  debug:
+    msg: "{{ users | community.general.to_prettytable('role', 'name', 'age') }}"
+
+- name: Display a table with selective column output (only show name and role fields)
+  debug:
+    msg: "{{ users | community.general.to_prettytable(column_order=['name', 'role']) }}"
+
+- name: Display a table with custom headers
+  debug:
+    msg: "{{ users | community.general.to_prettytable(header_names=['User Name', 'User Age', 'User Role']) }}"
+
+- name: Display a table with custom alignments
+  debug:
+    msg: "{{ users | community.general.to_prettytable(column_alignments={'name': 'center', 'age': 'right', 'role': 'left'}) }}"
+
+- name: Combine multiple options
+  debug:
+    msg: "{{ users | community.general.to_prettytable(
+        column_order=['role', 'name', 'age'],
+        header_names=['Position', 'Full Name', 'Years'],
+        column_alignments={'name': 'center', 'age': 'right', 'role': 'left'}) }}"
```

Comment:
Instead of using a multi-line string with quotes, it's better to use `>-`. I would indent it this way, to increase readability:
```suggestion
    msg: >-
      {{
        users | community.general.to_prettytable(
          column_order=['role', 'name', 'age'],
          header_names=['Position', 'Full Name', 'Years'],
          column_alignments={'name': 'center', 'age': 'right', 'role': 'left'},
        )
      }}
```

Reply:
Revised the example tasks' style to match the proposed variant.